timemachine/md/moves.py

```diff
@@ -12,69 +13,85 @@
 
 from timemachine import lib
 from timemachine.lib import custom_ops
-from timemachine.md.barostat.utils import get_bond_list, get_group_indices
 from timemachine.md.states import CoordsVelBox
-from timemachine.potentials import BoundPotential, HarmonicBond
+from timemachine.potentials import BoundPotential
 
 _State = TypeVar("_State")
 
 
-class MonteCarloMove(Generic[_State]):
-    n_proposed: int = 0
-    n_accepted: int = 0
+class Move(Generic[_State], ABC):
+    @abstractmethod
+    def move(self, _: _State) -> _State:
+        ...
+
+
+@dataclass
+class MonteCarloMove(Move[_State], ABC):
 
+    _n_proposed: int = field(default=0, init=False)
+    _n_accepted: int = field(default=0, init=False)
+
+    @abstractmethod
     def propose(self, x: _State) -> Tuple[_State, float]:
         """return proposed state and log acceptance probability"""
-        raise NotImplementedError
 
     def move(self, x: _State) -> _State:
         proposal, log_acceptance_probability = self.propose(x)
-        self.n_proposed += 1
+        self._n_proposed += 1
 
         alpha = np.random.rand()
         acceptance_probability = np.exp(log_acceptance_probability)
         if alpha < acceptance_probability:
-            self.n_accepted += 1
+            self._n_accepted += 1
             return proposal
         else:
             return x
 
     @property
-    def acceptance_fraction(self):
-        if self.n_proposed > 0:
-            return self.n_accepted / self.n_proposed
-        else:
-            return 0.0
+    def n_proposed(self) -> int:
+        return self._n_proposed
 
+    @property
+    def n_accepted(self) -> int:
+        return self._n_accepted
+
+    @property
+    def acceptance_fraction(self) -> float:
+        return self._n_accepted / self._n_proposed if self._n_proposed else np.nan
 
-class CompoundMove(MonteCarloMove[_State]):
-    def __init__(self, moves: List[MonteCarloMove]):
-        """Apply each of a list of moves in sequence"""
-        self.moves = moves
+
+class MetropolisHastingsMove(MonteCarloMove[_State], ABC):
+    @abstractmethod
+    def propose_with_dlogp(self, x: _State) -> Tuple[_State, float]:
+        "return proposed state and its unnormalized log probability"
```

Comment:
documentation: `dlogp = `
* `logp(proposed) - logp(x)`, or 
* unnormalized `logp(proposed)`?

(adding a quick usage example in tests would be ideal, e.g. RWMH in 1D)

Reply:
Added documentation and renamed to `dlogq` (`q` to indicate unnormalized by convention) in https://github.com/proteneer/timemachine/pull/1134/commits/03373e08f9f1e4c1d4f61501690c44f4fb49b0bf.

Added 1D RWMH test in https://github.com/proteneer/timemachine/pull/1134/commits/b8aebff39f3b1f814c373f79ee598e46fab96082.